### PR TITLE
Migrate kubernetes-external-secrets CRD to v1.22

### PR DESCRIPTION
**Pull request by chases2**
xref #3735

### CODE REVIEW

File: prow/cluster/kubernetes-external-secrets_crd.yaml [hunk 1]
```diff
@@ -1,6 +1,5 @@
 ---
-# From https://github.com/external-secrets/kubernetes-external-secrets/tree/f866e34e0319d9c54ea17d07f5d5818a28d9d0f6
```
Comment (cjwagner) [inline prow/cluster/kubernetes-external-secrets_crd.yaml:2]: Please include a line like this indicating where the new file was copied from.

Comment (chases2) [reply prow/cluster/kubernetes-external-secrets_crd.yaml:2]: It wasn't; that version doesn't seem to have a v1.22 compatible copy.

This was created by following https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22 instead

**Review by fejta — APPROVED** [review]
/hold

File: prow/cluster/kubernetes-external-secrets_crd.yaml (hunk 1 above)
Comment (cjwagner) [reply prow/cluster/kubernetes-external-secrets_crd.yaml:2]: What about this? https://github.com/external-secrets/kubernetes-external-secrets/blob/master/charts/kubernetes-external-secrets/crds/kubernetes-client.io_externalsecrets_crd.yaml

Comment (chaodaiG) [reply prow/cluster/kubernetes-external-secrets_crd.yaml:2]: The link posted by Cole should work, according to release notes https://github.com/external-secrets/kubernetes-external-secrets/releases/tag/8.0.1 has updated to crd/v1

Comment (chases2) [reply prow/cluster/kubernetes-external-secrets_crd.yaml:2]: Copied and added line

**Review by chaodaiG — APPROVED** [review]
/hold
might want to check release notes for breaking changes

**Pull request merged**